edge_prediction_experiment/01_hetionet_recall_dag.py

```diff
@@ -0,0 +1,348 @@
+            "pr_threshold": np.append(pr_threshold, 1),
+        }
+    )
+    >> ply.define(model=f'"pred_median/AUC={auc(recall, precision):.2f}"')
+)
+
+performance_map["AUROC"] = performance_map["AUROC"].append(
+    pd.DataFrame({"fpr": fpr, "tpr": tpr, "roc_threshold": roc_threshold})
+    >> ply.define(model=f'"pred_median/AUC={auc(fpr, tpr):.2f}"')
+)
+# -
+
+g = (
+    p9.ggplot(
+        performance_map["AUROC"]
+        >> ply.call(
+            ".append",
+            pd.DataFrame(
+                {
+                    "fpr": [0, 0.25, 0.5, 0.75, 1],
+                    "tpr": [0, 0.25, 0.5, 0.75, 1],
+                    "model": "random",
+                }
+            ),
+        )
+    )
+    + p9.aes(x="fpr", y="tpr", group="model", color="model", linetype="model")
+    + p9.geom_line()
+    + p9.theme_seaborn("white")
+    + p9.scale_color_manual(["#1b9e77", "#d95f02", "#7570b3", "#000000"])
+    + p9.scale_linetype_manual(["solid", "solid", "solid", "dashed"])
+)
+print(g)
+
+g = (
+    p9.ggplot(performance_map["PR"])
+    + p9.aes(x="recall", y="precision", group="model", color="model", linetype="model")
+    + p9.geom_line()
+    + p9.theme_seaborn("white")
+    + p9.scale_color_manual(["#1b9e77", "#d95f02", "#7570b3", "#000000"])
+    + p9.scale_linetype_manual(["solid", "solid", "solid", "dashed"])
+    + p9.geom_hline(
+        yintercept=(
+            test_entity_df >> ply.query("metric=='pred_median'") >> ply.pull("hetionet")
+        ).mean(),
+        linetype="dashed",
+    )
+)
+print(g)
+
+# ## Estimate number of new Edges Added
+
+# +
+df_iterator = (
+    performance_map["PR"]
+    >> ply.query("model.str.contains('max')& pr_threshold < 1")
+    >> ply.call(".round", {"pr_threshold": 2})
+    >> ply.distinct("pr_threshold", "last")
+)
+
+edges_added_records = []
+for idx, row in df_iterator.iterrows():
+    cutoff = row["pr_threshold"]
+
+    values_added = (
+        all_dag_df
+        >> ply.query("metric.str.contains('max')")
+        >> ply.query("score >= @cutoff")
+        >> ply.pull("hetionet")
+    )
+
+    edges_added_records.append(
+        {
+            "edges": values_added.sum(),
+            "in_hetionet": "Existing",
+            "precision": row["precision"],
+            "sen_cutoff": cutoff,
+        }
+    )
+
+    edges_added_records.append(
+        {
+            "edges": values_added.shape[0] - values_added.sum(),
+            "in_hetionet": "Novel",
+            "precision": row["precision"],
+            "sen_cutoff": cutoff,
+        }
+    )
+
+
+edges_added_df = pd.DataFrame.from_records(edges_added_records)
+edges_added_df >> ply.slice_rows(10)
+# -
+
+g = (
+    p9.ggplot(edges_added_df >> ply.query("edges > 0 & precision > 0"))
+    + p9.aes(x="precision", y="edges", color="in_hetionet")
+    + p9.geom_point()
+    + p9.theme_seaborn("white")
+    + p9.scale_color_brewer(type="qual", palette=2)
+)
+print(g)
+
+# +
+edges_df = pd.DataFrame.from_records(
```

Comment:
Is this the recall associated with the a certain precision value?

Reply:
>Is this the recall associated with the a certain precision value?

Good question. I didn't record the precision values here, but that will good to report. I'll add that before merging.